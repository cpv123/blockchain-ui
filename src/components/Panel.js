import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import { 
    Typography,
    ExpansionPanel, 
    ExpansionPanelSummary,
    ExpansionPanelDetails
} from '@material-ui/core';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

const styles = (theme) => ({
    rootGrey: {
        margin: 15,
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: 'lightGrey'
    },
    rootWhite: {
        ...styles.rootGrey,
        backgroundColor: 'white',
    },
    heading: {
        fontSize: theme.typography.pxToRem(15),
        fontWeight: 'bold',
    },
});

function Panel({ classes, title, children, color }) {
    return (
        <ExpansionPanel className={color === 'grey' ? classes.rootGrey : classes.rootWhite}>
            <ExpansionPanelSummary expandIcon={<ExpandMoreIcon />}>
                <Typography className={classes.heading}>{title}</Typography>
            </ExpansionPanelSummary>
            <ExpansionPanelDetails>
                {children}
            </ExpansionPanelDetails>
        </ExpansionPanel>
    );
}

Panel.propTypes = {
    classes: PropTypes.object.isRequired,
    title: PropTypes.string,
    children: PropTypes.any,
    color: PropTypes.string,
};

Panel.defaultProps = {
    color: 'grey',
}

export default withStyles(styles)(Panel);