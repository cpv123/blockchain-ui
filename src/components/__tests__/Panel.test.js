import React from 'react';
import { shallow } from 'enzyme';
import Panel from '../Panel';

const mockProps = {
    title: "Test Panel",
    children: {},
};

describe('the Panel component', () => {

    it('renders correctly', () => {
        const wrapper = shallow(<Panel {...mockProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders children correctly', () => {
        const wrapper = shallow(
            <Panel title={mockProps.title}>
                <div>
                    <p>Test panel children</p>
                </div>
            </Panel>
        );
        expect(wrapper.find("div")).toHaveLength(1);
        expect(wrapper.find("p")).toHaveLength(1);
        expect(wrapper.find("p").text()).toEqual("Test panel children");
    });

    it('renders the title prop correctly', () => {
        const wrapper = shallow(<Panel {...mockProps} />);
        expect(wrapper.prop("title")).toEqual("Test Panel");
    });

    it('handles no children', () => {
        const wrapper = shallow(<Panel title={mockProps.title} />);
        expect(wrapper.exists()).toBe(true);
    });
})

