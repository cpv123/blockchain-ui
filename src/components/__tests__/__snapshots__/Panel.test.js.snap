// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Panel component renders correctly 1`] = `
<Panel
  classes={
    Object {
      "heading": "Panel-heading-3",
      "rootGrey": "Panel-rootGrey-1",
      "rootWhite": "Panel-rootWhite-2",
    }
  }
  color="grey"
  title="Test Panel"
>
  <Component />
</Panel>
`;
